flink-runtime/src/main/java/org/apache/flink/runtime/util/BoundedFIFOQueue.java

```diff
@@ -102,11 +102,12 @@ private BoundedFIFOQueue(int maxSize, Iterable<T> initialElements, int initialEl
      * @throws NullPointerException If {@code null} is passed as an element.
      */
     public void add(T element) {
-        elementCount++;
-        elements.add(Preconditions.checkNotNull(element));
+        if (elements.add(Preconditions.checkNotNull(element))) {
```

Comment:
we don't really need the condition. Even if we were to use some other data-structure in the future where the insertion might fail, based on the current semantics of elementCount it should be incremented in any case.

Reply:
This was refactored as part of the `elementCount` revert